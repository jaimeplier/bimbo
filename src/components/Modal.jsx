import React, { Component } from 'react';
import ReactModal from 'simple-react-modal';

export default class Modal extends Component {
  render() {
    return (
      <ReactModal
        show={this.props.show}
        transitionSpeed={200}
        style={{ transition: 'opacity 0.2s ease-in', background: 'rgba(0, 0, 0, 0.5)' }}
        containerClassName="container"
        containerStyle={{ width: null, borderRadius: '2px' }}
        closeOnOuterClick
        onClose={this.props.onHide}
      >

        { this.props.children }
      </ReactModal>
    );
  }
}
