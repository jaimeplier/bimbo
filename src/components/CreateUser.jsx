import React, { Component } from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import Poly from './../utils/i18n';

import Modal from './Modal';

import Form from './Form';

export default class CreateUser extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isShown: false,
    };

    this.showModal = this.showModal.bind(this);
    this.hideModal = this.hideModal.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.createdUser = this.createdUser.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.copyToClipboard = this.copyToClipboard.bind(this);
  }

  showModal() {
    this.setState({ isShown: true });
  }
  hideModal() {
    this.setState({ isShown: false, success: false, user: {} });
  }

  handleInputChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }
  createdUser(user) {
    this.setState({ user, success: true });
  }

  handleSubmit() {
    if (this.state.isShown) {
      const { type } = this.props;
      this.setState({ errors: {} });
      if (!this.state.name) return this.setState({ errors: { name: 'Nombre requerido' } });
      if (
        type === 'manager' &&
        !this.state.email
      ) return this.setState({ errors: { email: 'Correo requerido' } });
      this.props.submit({
        name: this.state.name,
        email: this.state.email,
        language: 'es',
      }, this.createdUser);
      this.setState({ name: '', email: '' });
    }
  }

  copyToClipboard() {
    const textField = document.createElement('textarea');
    textField.innerText = this.state.user.accessPin;
    document.body.appendChild(textField);
    textField.select();
    document.execCommand('copy');
    textField.remove();
  }

  render() {
    return (
      <div className="create-user">
        <RaisedButton
          label={Poly.t('Create User')}
          primary
          onClick={this.showModal}
        />
        <Modal
          show={this.state.isShown}
          onHide={this.hideModal}
        >
          <div className="left modal-content card-padding" >
            <p className="card-title inline">{Poly.t('Create User')}</p>
            <button className="button grey-btn" onClick={this.hideModal}>X</button>
            { !this.state.success ?
              <Form onSubmit={this.handleSubmit}>
                <TextField
                  className="input"
                  name="name"
                  errorText={this.state.errors && this.state.errors.name}
                  onChange={this.handleInputChange}
                  floatingLabelText={Poly.t('Name')}
                />
                { this.props.type === 'manager' ?
                  <TextField
                    className="input"
                    name="email"
                    errorText={this.state.errors && this.state.errors.name}
                    onChange={this.handleInputChange}
                    floatingLabelText={Poly.t('Email')}
                  />
                : null }
                <div className="float-fix">
                  <RaisedButton
                    label={Poly.t('Create')}
                    primary
                    className="float-right"
                    type="submit"
                  />
                </div>
              </Form>
            :
              <div>
                <p>{Poly.t('Successfully created user')}</p>
                { this.props.type === 'factory' ?
                  <div>
                    <p>
                      {Poly.t('User PIN')}: {this.state.user.accessPin}
                    </p>
                    <RaisedButton
                      label={Poly.t('Copy to clipboard')}
                      primary
                      onClick={this.copyToClipboard}
                    />
                  </div>
                : null }
                { this.props.type === 'manager' ?
                  <p>
                    { Poly.t('Registration instructions sent to')}: { this.state.user.email }
                  </p>
                : null }
              </div>
            }
          </div>
        </Modal>
      </div>
    );
  }
}
