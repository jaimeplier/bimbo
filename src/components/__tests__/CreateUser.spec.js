import React from 'react';
import { shallow } from 'enzyme';
import CreateUser from '../CreateUser';
import Modal from '../Modal';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import SubmitButton from '../SubmitButton.jsx';

describe('CreateUser', () => {

  test('Component shuold render', () => {
    const createUser = shallow(<CreateUser />);
    expect(createUser).toMatchSnapshot();
  })

  test('Shows modal on button click', () => {
    const createUser = shallow(<CreateUser />);
    createUser.find(RaisedButton).simulate('click');

    expect(createUser.find(Modal).prop('show')).toBe(true);
  })

  test('Manager type should render email and name input field', () => {
    const createUser = shallow(<CreateUser type="manager" />);
    createUser.find(RaisedButton).simulate('click');

    expect(createUser.find(TextField)).toHaveLength(2);
    expect(createUser.find(TextField).at(0).prop('name')).toBe('name');
    expect(createUser.find(TextField).at(1).prop('name')).toBe('email');
  })

  test('Factory type should render only name input field', () => {
    const createUser = shallow(<CreateUser type="factory" />);
    createUser.find(RaisedButton).simulate('click');

    expect(createUser.find(TextField)).toHaveLength(1);
    expect(createUser.find(TextField).at(0).prop('name')).toBe('name');
  })

})


