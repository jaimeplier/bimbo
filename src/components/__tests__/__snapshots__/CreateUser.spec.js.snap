exports[`CreateUser Component shuold render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Create User"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false} />,
        <Modal
          onHide={[Function]}
          show={false}>
          <div
                    className="left modal-content card-padding">
                    <p
                              className="card-title inline">
                              Create User
                    </p>
                    <button
                              className="button grey-btn"
                              onClick={[Function]}>
                              X
                    </button>
                    <div>
                              <TextField
                                        className="input"
                                        disabled={false}
                                        errorText={undefined}
                                        floatingLabelFixed={false}
                                        floatingLabelText="Name"
                                        fullWidth={false}
                                        multiLine={false}
                                        name="name"
                                        onChange={[Function]}
                                        rows={1}
                                        type="text"
                                        underlineShow={true} />
                              <div
                                        className="float-fix">
                                        <CreateUser
                                                  className="float-right"
                                                  label="Create"
                                                  onFormSubmit={[Function]}
                                                  primary={true} />
                              </div>
                    </div>
          </div>
</Modal>,
      ],
      "className": "create-user",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "label": "Create User",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": true,
          "secondary": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="left modal-content card-padding">
            <p
                        className="card-title inline">
                        Create User
            </p>
            <button
                        className="button grey-btn"
                        onClick={[Function]}>
                        X
            </button>
            <div>
                        <TextField
                                    className="input"
                                    disabled={false}
                                    errorText={undefined}
                                    floatingLabelFixed={false}
                                    floatingLabelText="Name"
                                    fullWidth={false}
                                    multiLine={false}
                                    name="name"
                                    onChange={[Function]}
                                    rows={1}
                                    type="text"
                                    underlineShow={true} />
                        <div
                                    className="float-fix">
                                    <CreateUser
                                                className="float-right"
                                                label="Create"
                                                onFormSubmit={[Function]}
                                                primary={true} />
                        </div>
            </div>
</div>,
          "onHide": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="card-title inline">
                Create User
</p>,
              <button
                className="button grey-btn"
                onClick={[Function]}>
                X
</button>,
              <div>
                <TextField
                                className="input"
                                disabled={false}
                                errorText={undefined}
                                floatingLabelFixed={false}
                                floatingLabelText="Name"
                                fullWidth={false}
                                multiLine={false}
                                name="name"
                                onChange={[Function]}
                                rows={1}
                                type="text"
                                underlineShow={true} />
                <div
                                className="float-fix">
                                <CreateUser
                                                className="float-right"
                                                label="Create"
                                                onFormSubmit={[Function]}
                                                primary={true} />
                </div>
</div>,
            ],
            "className": "left modal-content card-padding",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create User",
                "className": "card-title inline",
              },
              "ref": null,
              "rendered": "Create User",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "X",
                "className": "button grey-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "X",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <TextField
                    className="input"
                    disabled={false}
                    errorText={undefined}
                    floatingLabelFixed={false}
                    floatingLabelText="Name"
                    fullWidth={false}
                    multiLine={false}
                    name="name"
                    onChange={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={true} />,
                  null,
                  <div
                    className="float-fix">
                    <CreateUser
                                        className="float-right"
                                        label="Create"
                                        onFormSubmit={[Function]}
                                        primary={true} />
</div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "input",
                    "disabled": false,
                    "errorText": undefined,
                    "floatingLabelFixed": false,
                    "floatingLabelText": "Name",
                    "fullWidth": false,
                    "multiLine": false,
                    "name": "name",
                    "onChange": [Function],
                    "rows": 1,
                    "type": "text",
                    "underlineShow": true,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <CreateUser
                      className="float-right"
                      label="Create"
                      onFormSubmit={[Function]}
                      primary={true} />,
                    "className": "float-fix",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "float-right",
                      "label": "Create",
                      "onFormSubmit": [Function],
                      "primary": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Create User"
            labelPosition="after"
            onClick={[Function]}
            primary={true}
            secondary={false} />,
          <Modal
            onHide={[Function]}
            show={false}>
            <div
                        className="left modal-content card-padding">
                        <p
                                    className="card-title inline">
                                    Create User
                        </p>
                        <button
                                    className="button grey-btn"
                                    onClick={[Function]}>
                                    X
                        </button>
                        <div>
                                    <TextField
                                                className="input"
                                                disabled={false}
                                                errorText={undefined}
                                                floatingLabelFixed={false}
                                                floatingLabelText="Name"
                                                fullWidth={false}
                                                multiLine={false}
                                                name="name"
                                                onChange={[Function]}
                                                rows={1}
                                                type="text"
                                                underlineShow={true} />
                                    <div
                                                className="float-fix">
                                                <CreateUser
                                                            className="float-right"
                                                            label="Create"
                                                            onFormSubmit={[Function]}
                                                            primary={true} />
                                    </div>
                        </div>
            </div>
</Modal>,
        ],
        "className": "create-user",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "label": "Create User",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": true,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="left modal-content card-padding">
              <p
                            className="card-title inline">
                            Create User
              </p>
              <button
                            className="button grey-btn"
                            onClick={[Function]}>
                            X
              </button>
              <div>
                            <TextField
                                          className="input"
                                          disabled={false}
                                          errorText={undefined}
                                          floatingLabelFixed={false}
                                          floatingLabelText="Name"
                                          fullWidth={false}
                                          multiLine={false}
                                          name="name"
                                          onChange={[Function]}
                                          rows={1}
                                          type="text"
                                          underlineShow={true} />
                            <div
                                          className="float-fix">
                                          <CreateUser
                                                        className="float-right"
                                                        label="Create"
                                                        onFormSubmit={[Function]}
                                                        primary={true} />
                            </div>
              </div>
</div>,
            "onHide": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="card-title inline">
                  Create User
</p>,
                <button
                  className="button grey-btn"
                  onClick={[Function]}>
                  X
</button>,
                <div>
                  <TextField
                                    className="input"
                                    disabled={false}
                                    errorText={undefined}
                                    floatingLabelFixed={false}
                                    floatingLabelText="Name"
                                    fullWidth={false}
                                    multiLine={false}
                                    name="name"
                                    onChange={[Function]}
                                    rows={1}
                                    type="text"
                                    underlineShow={true} />
                  <div
                                    className="float-fix">
                                    <CreateUser
                                                      className="float-right"
                                                      label="Create"
                                                      onFormSubmit={[Function]}
                                                      primary={true} />
                  </div>
</div>,
              ],
              "className": "left modal-content card-padding",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create User",
                  "className": "card-title inline",
                },
                "ref": null,
                "rendered": "Create User",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "X",
                  "className": "button grey-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "X",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <TextField
                      className="input"
                      disabled={false}
                      errorText={undefined}
                      floatingLabelFixed={false}
                      floatingLabelText="Name"
                      fullWidth={false}
                      multiLine={false}
                      name="name"
                      onChange={[Function]}
                      rows={1}
                      type="text"
                      underlineShow={true} />,
                    null,
                    <div
                      className="float-fix">
                      <CreateUser
                                            className="float-right"
                                            label="Create"
                                            onFormSubmit={[Function]}
                                            primary={true} />
</div>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "input",
                      "disabled": false,
                      "errorText": undefined,
                      "floatingLabelFixed": false,
                      "floatingLabelText": "Name",
                      "fullWidth": false,
                      "multiLine": false,
                      "name": "name",
                      "onChange": [Function],
                      "rows": 1,
                      "type": "text",
                      "underlineShow": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  null,
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <CreateUser
                        className="float-right"
                        label="Create"
                        onFormSubmit={[Function]}
                        primary={true} />,
                      "className": "float-fix",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "float-right",
                        "label": "Create",
                        "onFormSubmit": [Function],
                        "primary": true,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
